CRM-21840: Show Options Edit Link for Radio and Checkbox Groups

mickadoo opened the pull request:
Overview
----------------------------------------

A tool icon is provided to allow editing of option groups, however this icon is only shown for `select` input types. 

To make it easier to edit option groups this PR will also display the edit link for `radio` and `checkbox` input types.

Before
----------------------------------------

The edit link is only shown for `select` input types

![image](https://user-images.githubusercontent.com/6374064/37465927-64f3235a-2854-11e8-924d-a4546601b20f.png)

![image](https://user-images.githubusercontent.com/6374064/37465971-827d2b28-2854-11e8-80e1-f01ef8d5b6c7.png)


After
----------------------------------------

The edit link is also shown for radio groups.

![image](https://user-images.githubusercontent.com/6374064/37465871-413b19a4-2854-11e8-8f18-cbddda6e1096.png)

The same is true for radio options belonging to custom fields

![image](https://user-images.githubusercontent.com/6374064/37469772-f3e407ce-285d-11e8-9668-c9b0e1da58ab.png)

It is also shown for checkbox groups.

#### Radio button edit example

![peek 2018-03-19 15-29](https://user-images.githubusercontent.com/6374064/37656033-cdf18458-2c3e-11e8-9f83-1d211f0a3fe4.gif)

#### Checkbox group edit example

![peek 2018-03-20 12-58](https://user-images.githubusercontent.com/6374064/37655963-949d4ce6-2c3e-11e8-96c9-9d0bd10dcc07.gif)

---

 * [CRM-21840: Show Tool Icon For Radio Button Groups](https://issues.civicrm.org/jira/browse/CRM-21840)

colemanw commented:
@mickadoo the reason that only `<select>` elements were originally supported is because the javascript has to change the form element markup after the popup is saved. That's much easier to do with a select list. If you want to try tackling it, that code is here:
https://github.com/civicrm/civicrm-core/blob/master/js/crm.optionEdit.js

mickadoo commented:
@colemanw ah right, that makes sense. I've added a few changes to support refreshing radio options after saving. I'm no whiz with Javascript so prepare yourself. I've also added a GIF to the PR description showing it in action.

colemanw commented:
I'd suggest not putting that code in Common.js since it is not needed elsewhere. Just leave it in the `crm.optionEdit.js` file since that's the only place it's used.
Also, if we're bothering to do this for radios, the code should be identical for checkboxes.

mickadoo commented:
@colemanw I've added support for checkboxes, updated the PR description and added another gif showing checkbox options editing.

colemanw commented:
This looks great @mickadoo - thanks for the screencasts.
My one nitpick is that the js code could be more concise. The functions `refreshSelectOptions` `refreshRadioOptions` and `refreshCheckboxOptions` are identical except for one line, so that could be a param to a consolidated fn.

mickadoo commented:
@colemanw I did some refactoring in the last commit to remove duplicate code

colemanw commented:
@mickadoo final task in this PR would be to rebase & squash it into one commit. That keeps the commit history on the master branch clean, minus the WIP back & forth commits in the PR.

mickadoo commented:
@colemanw squashed into one commit, there was one civilint warning before this but hopefully that should be fixed now too.

colemanw commented:
I've reviewed the code and tested this.
